packages/nouns-contracts/test/descriptorV3.test.ts

```diff
@@ -0,0 +1,136 @@
+import chai from 'chai';
+import { solidity } from 'ethereum-waffle';
+import { NounsDescriptorV3 } from '../typechain';
+import ImageData from '../files/image-data-v2.json';
+import ProposalImages from '../files/proposal-test-traits.json';
+import { LongestPart } from './types';
+import { deployNounsDescriptorV3, populateDescriptorV2 } from './utils';
+import { ethers } from 'hardhat';
+
+chai.use(solidity);
+const { expect } = chai;
+
+const eyesred =
+  '0x000b17100703000624010006240300012402020265012401000124020202650524020202650324020202650224020001240202026501240100012402020265022402000124020202650124010001240202026501240300062401000624';
+
+const hiprose =
+  '0x000b1710070300062101000621030001210202022401210100012102020224052102020224032102020224052102020224032102020224022102000121020202240121010001210202022401210300062101000621';
+
+describe('NounsDescriptorV3', () => {
+  let nounsDescriptor: NounsDescriptorV3;
+  let snapshotId: number;
+
+  const glassesTraitIndex = 1;
+
+  const updateGlassesTraits = async () => {
+    const newEncodedCompressedTrait = ProposalImages[0].glasses?.encodedCompressed ?? '';
+    const decompressedLength = ProposalImages[0].glasses?.originalLength ?? 0;
+    const itemCount = ProposalImages[0].glasses?.itemCount ?? 0;
+
+    const glassesCountBefore = await nounsDescriptor.glassesCount();
+
+    console.log(`glassesCountBefore: ${glassesCountBefore}`);
+    await nounsDescriptor.updateGlasses(newEncodedCompressedTrait, decompressedLength, itemCount, {
+      gasLimit: 30000000,
+    });
+
+    const glassesCountAfter = await nounsDescriptor.glassesCount();
```

Comment:
@EricDevito I believe the descriptor code insists that the image count before and after be the same? -- if so, why not asset this equality here in the test?